### ISSUE TITLE
What command do I use in latex that ar5iv will turn into a link to an ancillary file attached to the paper?

### ISSUE THREAD

**Issue by Strilanc** [dginev/ar5iv#90]
One of the joys of HTML is the ability to add hyperlinks. One of the common things papers want to link to is their own ancillary files. For example, in a figure caption you might say "this figure was generated using data from ancillary file {{{data.csv}}}".

What is the magic word for making {{{data.csv}}} a hyperlink to the actual data.csv ancillary file on arXiv?

**Comment by dginev** [comment]
@Strilanc can I ask if there is existing support for this in the PDF flow? 

Or are we discussing a brand new feature that hasn't existed until today.

**Comment by dginev** [comment]
Alright, then I can safely say it will take _a while_ before ar5iv has support built in. I have actually help implement a feature like this before at a _different_ publishing platform (www.authorea.com), so I can safely say the technical piece is actually quite simple. 

Much easier to enable this than an advanced latex package.

But since it is new, there is an open-endedness that requires coordinating with the arXiv team, and timing it according to their preferences. No promises, let's keep the issue open and see if we get enough interested eyes on it.

**Comment by Strilanc** [comment]
Here's one possible recommendation: when an href's target is a relative path (e.g. no "https://") and it begins with "anc/" and the rest of the path is the path to one of the paper's ancillary files, change the href to actually directly link to that ancillary file for the current version of the paper.

This makes sense because ancillary files are identified by putting them in the "anc/" folder when uploading, and the "anc/" does appear in the existing link. It's just not strictly speaking actually a link relative to the paper's link, so it needs some massaging to be correct.

**Comment by Strilanc** [comment]
For example, in the paper https://ar5iv.org/abs/2103.02202 if I were to have an `\href{anc/random_circuit_64.stim}{example circuit}`, you would rewrite the target of the href from `anc/random_circuit_64.stim` to `https://arxiv.org/src/2103.02202v3/anc/random_circuit_64.stim`. This would also make sense to do in the PDFs.

**Comment by Strilanc** [comment]
Justification for doing this... I've had researchers ask me how to get at ancillary files that I mention in papers. If they could just click the link instead of having to find the list on the right hand side of the arXiv abstract page, that issue would never come up. And from a "we should actually look at each others' data if we want science to be reliable" perspective, any removal of trivial barriers that make that harder to do is an easy win.

**Comment by dginev** [comment]
I agree completely with your motivation, but fleshing out an official feature requires some focused development effort.

For most "real" scientific work, ancillary files could be orders of magnitude larger than the size of the paper itself. It is common nowadays in computer science to use different external solutions to host data (Zenodo, github, figshare, others), and to link to the external resources from the paper. 

I am not saying arXiv has to go either way, but it is certainly a complex question that involves resource management and building up some additional infrastructure, so there will be a need to consider what will work best for arXiv's circumstances. Not my call to make at the moment.

**Comment by Strilanc** [comment]
If they're using an external solution, it's already possible to link to them using an `\href`. The tricky bit with the arXiv is that it generates an identifier *after I submit*, so I can't predict the URL where the ancillary files will be. I suppose I can get the same ultimate effect by doing a two step dance where the first version is just to nail down the identifier, and later versions can then explicitly use that identifier in the source code when defining hrefs.

**Comment by quantumgc** [comment]
It's worth noting that the ArXiV explicitly [recommends against](https://info.arxiv.org/help/ancillary_files.html#:~:text=Pointers%20inside%20your%20codebase%20should%20not%20reference%20this%20directory%20(linking%20may%20be%20lost%20upon%20announcement).) linking to ancillary files from within the article codebase.

**Comment by dginev** [comment]
@quantumgc Sure, but let's also note those docs target the pre-HTML state. And most of the discussion here faces the future, more so than the present.
